Add beta support of Safari

hobinjk opened the pull request:
Note that this uses a copy of all extension files instead of using a build step to copy them. This also uses a squashed version of the commits, I can unsquash if desired.

0xazure commented on New XKit.safariextension/Info.plist:
```diff
@@ -3,23 +3,23 @@
+	<string>New XKit</string>
 	<key>CFBundleIdentifier</key>
-	<string>com.studioxenix.xkit</string>
+	<string>com.xkitplus.xkit</string>
```
`xkitplus` vs `new[-]xkit`?

nightpool commented:
Unsquashd would probably be better for commit history purposes.

Are there any safari-specific diffs or are these straight copies?
On Fri, Oct 2, 2015 at 11:37 AM James Hobin <notifications@github.com>
wrote:

> Note that this uses a copy of all extension files instead of using a build
> step to copy them. This also uses a squashed version of the commits, I can
> unsquash if desired.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/new-xkit/XKit/pull/574
> Commit Summary
>
>    - Add beta support of Safari
>
> File Changes
>
>    - *R* New XKit.safariextension/Info.plist
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-0> (37)
>    - *A* New XKit.safariextension/Settings.plist
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-1> (5)
>    - *R* New XKit.safariextension/bridge.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-2> (169)
>    - *A* New XKit.safariextension/editor.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-3> (451)
>    - *A* New XKit.safariextension/global.html
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-4> (12)
>    - *A* New XKit.safariextension/global.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-5> (118)
>    - *R* New XKit.safariextension/icon.png
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-6> (0)
>    - *A* New XKit.safariextension/jquery.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-7> (4)
>    - *A* New XKit.safariextension/moment.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-8> (12)
>    - *A* New XKit.safariextension/nano.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-9> (752)
>    - *A* New XKit.safariextension/tiptip.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-10> (21)
>    - *A* New XKit.safariextension/xkit.css
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-11> (824)
>    - *A* New XKit.safariextension/xkit.js
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-12> (1323)
>    - *D* Safari/global.html
>    <https://github.com/new-xkit/XKit/pull/574/files#diff-13> (203)
>
> Patch Links:
>
>    - https://github.com/new-xkit/XKit/pull/574.patch
>    - https://github.com/new-xkit/XKit/pull/574.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/new-xkit/XKit/pull/574>.
>

0xazure commented:
It's probably safe to remove the extension file copies, unless you want to leave them here and remove them in whatever PR closes #546.

Also, the XKit Editor doesn't seem to load on the Safari version and I'm not entirely clear why.  I didn't notice any obvious error messages in the console, but it might have something to do with the Ace editor?

Wolvan commented:
@0xazure mind if I take a crack at the build process?

hobinjk commented:
@nightpool There are not actually that many changes, I think I will leave it squashed for now. It's only Rewrite global.js, Fix Whitespace Errors, Update/Fix Info.plist and then some miscellaneous bugfixes.

@0xazure I agree, I've removed the copies and renamed the folder.

0xazure commented on Safari/Info.plist:
```diff
@@ -3,23 +3,23 @@
+	<string>New XKit</string>
 	<key>CFBundleIdentifier</key>
-	<string>com.studioxenix.xkit</string>
+	<string>com.xkitplus.xkit</string>
```
Rebase made GH hid my comment, so here it is again:

`xkitplus` vs `new[-]xkit`?

hobinjk replied:
Oh right, sorry for not addressing this. I don't think we own new-xkit.com or newxkit.com, but someone has xkitplus.com. That is the rational for this identifier.

0xazure replied:
@ChuckL has https://portal.new-xkit.com/#/login for the XCloud login.

Wolvan replied:
We, or rather @ChuckL, does in fact own new-xkit.com

hobinjk replied:
Great, pointing it there.

0xazure commented:
I'm still getting issues with

```sh
XBackground Error: TypeError: undefined is not an object (evaluating 'ev.target.page.dispatchMessage')
```

on certain blogs.  This is a major blocker for us to release the Safari version, especially since it produces a whole chain of pop-up error messages.

0xazure commented:
In attempting to test out #571 I've been running into a lot of 

```sh
XBridge: Warning! Callback for HTTP Request lIQ5Ox1fICmjfIdC608eUs0EpenzGFBdwrg4e2tObd3cqB9CxE not found!
```

errors.  I'm not _quite_ sure that this means, but I do know that I cannot properly update any of the XKit extensions.  XKit barely installs itself correctly most of the time.  

I've set XKit to connect locally to `gulp server`, but it keeps having a `Server error, falling back for download of xkit_preferences.json`, and so falls back to the outdated distribution.  Near as I can tell this is a server error at the `xmlhttpRequest` level and not a 'server error' as identified by XKit itself, but there's no difference between the setup for Safari and the one that works on Chrome so I'm not sure what's causing the error.  I have, of course, verified that the self-signed cert is accepted by Safari and is in the cert store.

hobinjk commented:
Updated with more functionality, I no longer see errors on the discussed page, but I also didn't specifically try to fix those errors.

hobinjk commented:
https://files.gitter.im/new-xkit/XKit/taHc/New-XKit.safariextz is the latest build

nightpool commented:
lol at sharing files over gitter haha

On Sun, Oct 11, 2015 at 8:54 PM James Hobin <notifications@github.com>
wrote:

> https://files.gitter.im/new-xkit/XKit/taHc/New-XKit.safariextz is the
> latest build
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/new-xkit/XKit/pull/574#issuecomment-147267627>.
>

0xazure commented:
I'll review this when I have time, hopefully late tomorrow.  I'll take another stab at serving stuff locally and see if I can narrow the issue there down further, if it isn't already resolved.

0xazure commented:
I can confirm that the issue with

```sh
'ev.target.page.dispatchMessage'
```

has been resolved and Safari appears to be working as intended.

However, I am still unable to test extensions on Safari when serving them locally.  As can be seen below, 
`SafariExtensionMessageEvent` is returning a `status` of 0, even though the local server is reporting it served the requested asset with `status` 200.

```sh
{request: Object, status: 0, settings: undefined, request_id: "b5paRapSkhec1lakt5vaPIsWPmuB0YtIYDgQyaPYx2mlmTxl91", headers: ""}
```

This explains all of the 

```
XBridge: Warning! Callback for HTTP Request lIQ5Ox1fICmjfIdC608eUs0EpenzGFBdwrg4e2tObd3cqB9CxE not found!
```

messages, as the returned message object has no callbacks to be called.

Note that everything is working as expected when Safari sends requests to our `gh-pages` URL, so this seems to be an issue with SSL certificates even though my Safari installation is correctly configured to trust the dev cert as can be seen below:

![screen shot 2015-10-12 at 20 17 44](https://cloud.githubusercontent.com/assets/1041456/10442248/553a5b58-711e-11e5-87b6-dbfa7163c389.png)

I am of the opinion that we can track down the issue(s) with serving assets locally at a later date, so I feel comfortable merging this now.

@hobinjk anything more you want to do with this branch before I merge?

hobinjk commented:
Great! I also could not figure out how to serve locally. I would guess that it is some form of extension security nonsense. Merge whenever you want to.

0xazure commented:
@homu r+

homu commented:
:pushpin: Commit ec855fc has been approved by `0xazure`

<!-- @homu r=0xazure ec855fc17bd3269641178af662f758bfadb39247 -->

homu commented:
:hourglass: Testing commit ec855fc with merge 989b5af...

homu commented:
:sunny: Test successful - [status](https://travis-ci.org/new-xkit/XKit/builds/85029049)
